js-envs/src/main/scala/org/scalajs/jsenv/Utils.scala

```diff
@@ -27,12 +27,91 @@ private[jsenv] object Utils {
       else deadline.isOverdue
   }
 
-  object OptDeadline {
+  private[jsenv] object OptDeadline {
     def apply(timeout: Duration): OptDeadline = {
       new OptDeadline(timeout match {
         case timeout: FiniteDuration => timeout.fromNow
         case _                       => null
       })
     }
   }
+
+  // !!! BEGIN COPY-PASTED CODE with ir/.../Utils.scala
+
+  private final val EscapeJSChars = "\\b\\t\\n\\v\\f\\r\\\"\\\\"
+
+  def escapeJS(str: String): String = {
```

Comment:
This shouldn't be in the public interface of `jsEnvs` :-/ But I don't have a better solution for this ATM. Note that `OptDeadline` will vanish completely with #3033 (for which I hope to send out a PR today).

Reply:
Looking at the next commit, maybe we should just move this to `io` as well? Neither `URIUtils` nor this are `io`, but essentially "everything" needs it.

Reply:
We cannot move it to `io` because it is used by `ir.Printers`.

Reply:
I'm OK with the duplication in IR. I'm mostly concerned about it being part of the interface of `jsEnvs`.

Reply:
Ah.

OK maybe we can put it in `io.JSUtils` then, and make the other one `private[ir]`.

Reply:
SG.